docs/csharp/language-reference/builtin-types/reference-types.md

```diff
@@ -0,0 +1,178 @@
+
+The following example contrasts a variable of type `dynamic` to a variable of type `object`. To verify the type of each variable at compile time, place the mouse pointer over `dyn` or `obj` in the `WriteLine` statements. IntelliSense shows **dynamic** for `dyn` and **object** for `obj`.
+
+[!code-csharp[csrefKeywordsTypes#21](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/dynamic1.cs#21)]
+
+The `WriteLine` statements display the run-time types of `dyn` and `obj`. At that point, both have the same type, integer. The following output is produced:
+
+```console
+System.Int32
+System.Int32
+```
+
+To see the difference between `dyn` and `obj` at compile time, add the following two lines between the declarations and the `WriteLine` statements in the previous example.
+
+```csharp
+dyn = dyn + 3;
+obj = obj + 3;
+```
+
+ A compiler error is reported for the attempted addition of an integer and an object in expression `obj + 3`. However, no error is reported for `dyn + 3`. The expression that contains `dyn` is not checked at compile time because the type of `dyn` is `dynamic`.
+
+The following example uses `dynamic` in several declarations. The `Main` method also contrasts compile-time type checking with run-time type checking.
+
+[!code-csharp[csrefKeywordsTypes#25](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsTypes/CS/dynamic2.cs#25)]
+
+## The string type
+
+The `string` type represents a sequence of zero or more Unicode characters. `string` is an alias for <xref:System.String?displayProperty=nameWithType> in .NET.
+
+Although `string` is a reference type, the equality operators (`==` and `!=`) are defined to compare the values of `string` objects, not references. This makes testing for string equality more intuitive. For example:
+
+```csharp
+string a = "hello";
+string b = "h";
+// Append to contents of 'b'
+b += "ello";
+Console.WriteLine(a == b);
+Console.WriteLine((object)a == (object)b);
+```
+
+This displays "true" and then "false" because the content of the strings are equivalent, but `a` and `b` do not refer to the same string instance.
+
+The + operator concatenates strings:
+
+```csharp
+string a = "good " + "morning";
+```
+
+This creates a string object that contains "good morning".
+
+Strings are *immutable*--the contents of a string object cannot be changed after the object is created, although the syntax makes it appear as if you can do this. For example, when you write this code, the compiler actually creates a new string object to hold the new sequence of characters, and that new object is assigned to b. The string "h" is then eligible for garbage collection.
```

Comment:
I'm not sure that "h" is eligible for garbage collection since it's a string in the intern pool, and I'm not sure whether the intern pool is garbage collected or not. @Maoni0 can confirm. What certainly is garbage collected is the original memory allocated to `b`.

For example, the following shows that "h" is still in the intern pool:
```
            string str = "h";
            str = null;
            Thread.Sleep(10000);
            GC.Collect();
            string str2 = "hello";
            str2 = str2.Substring(0, 1);
            Console.WriteLine($"'{str2}' interned: {String.IsInterned(str2) != null}");
```